tests/functional/n/name/name_is_parameter_and_nonlocal.py

```diff
@@ -0,0 +1,6 @@
+# pylint: disable=missing-docstring,invalid-name,nonlocal-without-binding
+
+
+def tomato(is_tasty: bool = True):
+    nonlocal color
+    nonlocal is_tasty # [name-is-parameter-and-nonlocal]
```

Comment:
```suggestion
    nonlocal is_tasty # [name-is-parameter-and-nonlocal]


def tomato(*is_tasty):
    nonlocal is_tasty # [name-is-parameter-and-nonlocal]
```

I think this would also raise a syntax error, right?

Reply:
I guess for that I can do something like
```python
        node_arg_names = {arg.name for arg in node.args.args}
        for local_name, local_nodes in node.locals.items():
            for local_node in local_nodes:
                if isinstance(local_node, nodes.Arguments):
                    node_arg_names.add(local_name)
```

Not sure if there's an easier way to retrieve locals that are considered arguments.

Reply:
For nested scopes and functions, are you looking for something like this?
```python
def grape(is_tasty: bool = True):
    nonlocal is_tasty # [name-is-parameter-and-nonlocal]

    def grape_juice(is_tasty: bool = True):
        nonlocal is_tasty  # [name-is-parameter-and-nonlocal]

    grape_juice()
```

Reply:
Doesn't `node.args.arguments()` work? According to the docstring that should be able to return all names that are used as parameters.

Reply:
Yeah, and variants of it. I haven't tested this out locally to see which combinations return a syntax error, but I think it is good to add some (regression tests) for possible combinations:
- outer scope with nonlocal inner scope with nonlocal
- outer scope without nonlocal inner scope with nonlocal
- outer scope with nonlocal inner scope without nonlocal
etc.

Scoping rules can create annoying false positives for messages like this.

Reply:
> Doesn't `node.args.arguments()` work? According to the docstring that should be able to return all names that are used as parameters.

When I inspect `node.args` for a function that uses `(*is_tasty)`, it comes pretty much empty:
```python
(Pdb) vars(node.args)
{'lineno': None, 'col_offset': None, 'parent': <FunctionDef.watermelon l.9 at 0x7efe6e8078e0>, 'end_lineno': None, 'end_col_offset': None, 'position': None, 'vararg': 'is_tasty', 'kwarg': None, 'posonlyargs': [], 'posonlyargs_annotations': [], 'kwonlyargs_annotations': [], 'type_comment_args': [], 'type_comment_kwonlyargs': [], 'type_comment_posonlyargs': [], 'varargannotation': None, 'kwargannotation': None, 'args': [], 'defaults': [], 'kwonlyargs': [], 'kw_defaults': [], 'annotations': [], 'fromlineno': 9, 'tolineno': 9, 'arguments': []}
(Pdb) node.args.arguments
[]
```

Reply:
Sure! I can attempt that later today, this might also be related https://github.com/PyCQA/astroid/issues/761 - but not totally sure.

Reply:
Indeed let's not block this on too much refactoring and fix stuff as we run into them. I think this is a very clear use-case for a potential fix and I'd be happy to help release a patch version of `astroid` for it. I think there are other issues waiting on a patch as well, a regression in dataclasses for example.